tools/proofers/preview.js

```diff
@@ -1125,6 +1065,57 @@ $(function () {
 
         // ok true if no errors which would cause showstyle() or reWrap() to fail
         let ok = (issues === 0);
+
+        // make texts to be inserted to mark issues
+        let issueStarts = [];
+        let issueEnds = [];
+        let errorString;
+        issArray.forEach(function(issue) {
+            if (issue.type === 0) {
+                errorString = makeErrStr("hlt");
+            } else {
+                errorString = makeErrStr("err");
+            }
+            let message = previewMessages[issue.code].replace("%s", issue.subText);
```

Comment:
do we need to make sure issue.subText doesn't have any of the patterns used by replace (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#specifying_a_string_as_a_parameter). We ran into an issue like this in the find replace dialog, so you can see my solution there if necessary https://github.com/DistributedProofreaders/dproofreaders/blob/master/tools/proofers/srchrep.js#L27

Reply:
May not be an adequate test, but I used 

```
[** [386]][** [** [386]]$1\n\n\\n\\n$$2][** A][** B]
```
as proofers' notes, and what was shown in Preview was

```
[** B][** A][** [** [386]]$1\n\n\n\n$$2][** [386]]
```

I see two issues immediately 
* while the `$$` was accurately displayed in the preview, the `\\` was not. A single backslash appears to be echoed just fine, but if it's being used preceding another backslash or a `$`, it's treated as an escape character.
* There are four consecutive notes there -- they're displayed in the preview in reverse order.

Reply:
@chrismiceli The string is always predefined so it shouldn't be a problem. i.e. '/*', '/#', '*/', '#/', 'Footnote', 'Illustration', 'Sidenote', '&lt;tb&gt;'.

@srjfoo Interesting. I'll have to look into this.

Reply:
@srjfoo I've revised the sorting code so the notes appear in the correct order.
It looks like the backslash issue is caused by MathJax. If I disable the Math preview the backslashes in your example appear correctly. Does it work like that for you?

Reply:
Yep, the sorting looks good now.

And yes, turning off Math preview takes care of the backslash issue -- I didn't think of that. Is it fair to assume that that's something that will have to be lived with? (Bear in mind that I was using test strings that are unlikely to be used in reality to test per Chris' question).

Would it be appropriate to move the Math checkbox out of the "Configure" block and into the main Preview Toolbox? I realize we need to keep that small, but I had consciously decided "Well, there's no harm in leaving the Math checkbox checked permanently" and gone off and forgotten about it. If the checkbox is in the main toolbar, it's a reminder that "this is something you may want immediate on-off access to".

Reply:
> Yep, the sorting looks good now.
> 
> And yes, turning off Math preview takes care of the backslash issue -- I didn't think of that. Is it fair to assume that that's something that will have to be lived with? (Bear in mind that I was using test strings that are unlikely to be used in reality to test per Chris' question).
> 
> Would it be appropriate to move the Math checkbox out of the "Configure" block and into the main Preview Toolbox? I realize we need to keep that small, but I had consciously decided "Well, there's no harm in leaving the Math checkbox checked permanently" and gone off and forgotten about it. If the checkbox is in the main toolbar, it's a reminder that "this is something you may want immediate on-off access to".

I don't see why MathJax should need to change the backslash characters in this way. I'll enquire why it happens.
In the meantime we can live with it as we have been without noticing the problem. Moving the checkbox is something we could consider.